### PR TITLE
Fully integrated BE into auth process

### PR DESCRIPTION
**Pull request by akhtariev**
- INSERT query (not satisfied because need an inteface for the user to insert)
To make it work, I'll add an interface to add the allergies or surgeries
- SELECT query

Authentication is now complete. Please create your a new account. YOUR OLD ACCOUNTS WILL NOT WORK

### CODE REVIEW

File: client/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "version": "0.2.0",
+    "configurations": [
```
Comment (akhtariev) [inline client/.vscode/launch.json:3]: this stuff is for debugger

File: client/src/components/AuthenticatedApp.jsx [hunk 2]
```diff
@@ -22,13 +22,9 @@ const AuthenticatedApp = () => {
   const classes = useStyles();
   const dispatch = useDispatch();
   const { getTokenSilently, user } = useAuth0();
-  const userState = useSelector(state => state.user);
+  const patientState = useSelector(state => state.patients);
 
-  if (userState === undefined) {
-    return <TextResponse heading='Technical issue...' body='Please contact administrator' />;
```
Comment (akhtariev) [inline client/src/components/AuthenticatedApp.jsx:28]: This was here just to prevent unrecoverable errors

File: server/.vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "type": "node",
+            "request": "launch",
+            "name": "Launch Program",
```
Comment (akhtariev) [inline server/.vscode/launch.json:10]: Debugging stuff

File: server/sql/v1.sql [hunk 4]
```diff
@@ -1,3 +1,27 @@
+-- Change Log:
+-- Roman - 2020-06-13
+-- Medical_History - change type of id to SERIAL from CHAR
+-- Patient - change type of id to SERIAL from CHAR
```
Comment (akhtariev) [inline server/sql/v1.sql:4]: Update this, not serial -> CHAR(24) 24 refers to the size of the object id

File: server/src/routes/user/createUser.js [hunk 5]
```diff
@@ -1,8 +1,26 @@
...
+const logger = require('../../utils/logger');
 
-router.post("/:id", (req, res) => {
+router.post("/patient/:id", async (req, res) => {
+    // Create medical history
+    try {
+        const medHisResponse = await sqlClient.query(`INSERT INTO Medical_History (guardian_name, height, weight, date_created) VALUES ('', 0, 0, to_timestamp(${Date.now()} / 1000.0)) RETURNING *`);
+        if (medHisResponse) {
+            const medHisId = medHisResponse.rows[0].id;
+            const result = await sqlClient.query(`INSERT INTO Patient (id, mid, email, password, full_name, profile_picture_url) VALUES ('${req.params.id}', ${medHisId}, '${req.body.username}', '', '', '') RETURNING *`);
```
Comment (RachitMalik12) [inline server/src/routes/user/createUser.js:11]: shouldn't we take input from the user for weight, height in Medical history and patient name, profile pic url and password? I saw you were passing profile pic url from the client but you are not using it here. 
 If password in handled by library leave it but others we should take in as req.body as well. Maybe a quick form for the patient to enter their medical history details as soon as they sign up? we shouldnt insert '' for everything. Feel free to do the FE in another PR but we should take it from req body.

Comment (akhtariev) [reply server/src/routes/user/createUser.js:11]: This will be changed in another PR

Comment (akhtariev) [reply server/src/routes/user/createUser.js:11]: And you don't have any info about the profile picture when you're creating the user

Comment (akhtariev) [reply server/src/routes/user/createUser.js:11]: That data will be updated separately

Comment (RachitMalik12) [reply server/src/routes/user/createUser.js:11]: cool sounds good. Ill approve then lets get this in

**Pull request merged**